mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -40,18 +41,37 @@ export interface GraphObject {
 export interface GraphParams {
     graphType: GraphType,
     xAxisColumnIDs: ColumnID[],
-    yAxisColumnIDs: ColumnID[]
+    yAxisColumnIDs: ColumnID[],
+    safetyFilter: boolean
 }
 
 // Millisecond delay between loading graphs, so that
 // we don't load to many graphs when the user is clicking around
 const LOAD_GRAPH_TIMEOUT = 1000;
 
+// Graphing a dataframe with more than this number of rows will
+// give the user the option to apply the safety filter
+// Note: This must be kept in sync with the graphing heuristic in the mitosheet/graph folder
+const GRAPH_SAFETY_FILTER_CUTOFF = 1000;
+
+// Tooltips used to explain the Safety filter toggle
+const SAFETY_FILTER_DISABLED_MESSAGE = `Because you’re graphing less than ${GRAPH_SAFETY_FILTER_CUTOFF} rows of data, you can safely graph your data without applying a filter first.`
+const SAFETY_FILTER_ENABLED_MESSAGE = `Turning on Filter to Safe Size only graphs the first ${GRAPH_SAFETY_FILTER_CUTOFF} rows of your dataframe, ensuring that your browser tab won’t crash. Turning it off, graphs the entire dataframe and might crash your browser tab.`
```

Comment:
> Turning it off, graphs the entire dataframe and might crash your browser tab.

Turning Off the Filter to Safe Size graphs the entire data frame and may slow or crash your browser tab.